File: src/graph/shape/StencilRegistry.java
package graph.shape;

import java.util.HashMap;
import java.util.Map;

public class StencilRegistry
{
	/**
	 * 
	 */
	protected static Map<String, Stencil> _stencils = new HashMap<String, Stencil>();

	/**
	 * Adds the given stencil.
	 */
	public static void addStencil(String name, Stencil stencil)
	{
		_stencils.put(name, stencil);
	}

	/**
	 * Returns the stencil for the given name.
	 */
	public static Stencil getStencil(String name)
	{
		return _stencils.get(name);
	}

}

File: lib/shape/stencil_registry.dart
part of graph.shape;


class StencilRegistry {
  static Map<String, Stencil> _stencils = new HashMap<String, Stencil>();

  /**
   * Adds the given stencil.
   */
  static void addStencil(String name, Stencil stencil) {
    _stencils[name] = stencil;
  }

  /**
   * Returns the stencil for the given name.
   */
  static Stencil getStencil(String name) {
    return _stencils[name];
  }

}
